src/scancode/plugin_license_policy.py

```diff
@@ -0,0 +1,109 @@
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  ScanCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  ScanCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from os.path import abspath
+from os.path import dirname
+from os.path import exists
+from os.path import isdir
+from os.path import join
+
+import attr
+
+from commoncode import saneyaml
+from plugincode.post_scan import PostScanPlugin
+from plugincode.post_scan import post_scan_impl
+from scancode import CommandLineOption
+from scancode import POST_SCAN_GROUP
+
+
+@post_scan_impl
+class LicensePolicy(PostScanPlugin):
+    """
+    Add the "license_policy" attribute to a resouce if it contains a
+    detected license key that is found in the license_policy.yml file
+    """
+
+    attributes = dict(license_policy=attr.ib(default=attr.Factory(dict)))
+
+    sort_order = 9
+
+    options = [
+        CommandLineOption(('--license-policy',),
+            multiple=False,
+            metavar='<license_policy>',
+            help='Load and a License Policy file and apply it to the codebase at the '
+                 'Resource level.',
+            help_group=POST_SCAN_GROUP)
+    ]
+
+    def is_enabled(self, license_policy, **kwargs):
+        return license_policy
+
+    def process_codebase(self, codebase, license_policy, **kwargs):
+        """
+        Populate a license_policy mapping with three attributes: label, icon, 
+        and color_code at the File Resource level.
+        """
+        if not self.is_enabled(license_policy):
+            return
+
+        # load a dictionary of license_policies from a file
+        license_policies = load_license_policy(license_policy).get('license_policies')
+
+        # apply policy to Resources if they contain an offending license
+        for resource in codebase.walk(topdown=True):
+            if not resource.is_file:
+                continue
+
+            try:
+                resource_license_keys = [entry.get('key') for entry in resource.licenses]
+
+                for key in resource_license_keys:
+                    if key in license_policies.keys():
+                        # Apply the policy to the Resource
+                        resource.license_policy = license_policies[key]
+                        codebase.save_resource(resource)
+
+            except AttributeError:
+                # add license_policy regardless if there is --license info or not
```

Comment:
I was about to say that another way could have been to require a license scan... But this is actually better as this always works!

Reply:
I think we are could have this both as a separate plugin and we should be able to run on a "virtual" codebase alright as it is